apps/dashboard/src/contract-ui/tabs/overview/components/MarketplaceDetails.tsx

```diff
@@ -53,60 +48,40 @@ type ListingData =
     });
 
 type MarketplaceDetailsProps = {
-  contractAddress: string;
-  contractType: "marketplace" | "marketplace-v3";
+  contract: ThirdwebContract;
   features: string[];
   trackingCategory: TrackedLinkProps["category"];
 };
 
-interface MarketplaceDetailsVersionProps<T> {
-  contract: T;
-  trackingCategory: TrackedLinkProps["category"];
-  features: MarketplaceDetailsProps["features"];
-}
-
 export const MarketplaceDetails: React.FC<MarketplaceDetailsProps> = ({
-  contractAddress,
-  contractType,
+  contract,
   trackingCategory,
   features,
 }) => {
-  const { contract } = useContract(contractAddress, contractType);
-
-  if (contractType === "marketplace" && contract) {
-    // no longer supported
-    return null;
-  }
   return (
     <MarketplaceV3Details
-      contract={contract as MarketplaceV3}
+      contract={contract}
       trackingCategory={trackingCategory}
       features={features}
     />
   );
 };
 
 type ListingCardsSectionProps = {
-  contract: MarketplaceV3;
+  contract: ThirdwebContract;
   trackingCategory: TrackedLinkProps["category"];
 };
 
 const DirectListingCards: React.FC<ListingCardsSectionProps> = ({
   trackingCategory,
-  contract: v4Contract,
+  contract,
 }) => {
-  const chain = useV5DashboardChain(v4Contract.chainId);
-  const contract = getContract({
-    client: thirdwebClient,
-    address: v4Contract.getAddress(),
-    chain: chain,
-  });
   const directListingsHref = useTabHref("direct-listings");
   const countQuery = useReadContract(totalListings, { contract });
   const listingsQuery = useReadContract(getAllListings, {
     contract,
     count: 3n,
-    start: Math.max(BigNumber.from(countQuery?.data || 3)?.toNumber() - 3, 0),
+    start: Math.max(Number(countQuery?.data || 3n) - 3, 0),
```

Comment:
this is fine because it's always going to be a small enough number, however we do have a `max()` function in `v5` sdk that does `math.max` for bigints - u can use that here (export it if u need) to save some characters